ddtrace/contrib/redis_utils.py

```diff
@@ -71,7 +71,7 @@ async def _run_redis_command_async(ctx: core.ExecutionContext, func, args, kwarg
     try:
         result = await func(*args, **kwargs)
         return result
-    except Exception:
+    except BaseException:
```

Comment:
I'm not sure how this fixes the issue given that `BaseException` is a superclass of `UnboundLocalError`
```
>>> isinstance(UnboundLocalError(), BaseException)
True
```

Reply:
It's `asyncio.CancelledError` that is raised within the `try/except`, not `UnboundLocalError`.
The fixes will catch it and the `UnboundLocalError` will not be raised inside the `except`.

The test proves it works.